src/electionguard_gui/services/decryption_stages/decryption_s1_join_service.py

```diff
@@ -0,0 +1,45 @@
+from pymongo.database import Database
+from electionguard.ballot import BallotBoxState
+
+from electionguard_gui.models.decryption_dto import DecryptionDto
+from electionguard_gui.services.decryption_stages.decryption_stage_base import (
+    DecryptionStageBase,
+    get_tally,
+)
+
+
+class DecryptionS1JoinService(DecryptionStageBase):
```

Comment:
DecryptionS1JoinService and DecryptionS2AnnounceService are the meat of this PR, this is where the guardians calculate tally and spoiled ballot decryption shares and the guardian turns them into plaintext.